### PR TITLE
Gcc 8+ and 11 fixes

### PR DESCRIPTION
**Pull request by SpamapS**
While testing gearmand with newer versions of Ubuntu (20.04 and 22.04) I found a few minor things that needed updating in order to build and test.

### CODE REVIEW

File: libtest/main.cc [hunk 1]
```diff
@@ -332,7 +332,9 @@ int main(int argc, char *argv[], char* environ_[])
     do
     {
       exit_code= EXIT_SUCCESS;
-      fatal_assert(sigignore(SIGPIPE) == 0);
+      struct sigaction sa;
+      sa.sa_handler=  SIG_IGN;
```
Comment (esabol) [inline libtest/main.cc:336]: There should only be one space after the equals sign on this line, I think. Not that it really matters....

**Comment by esabol** [comment]
Also, please don't release a new gearmand without looking at PR #348. 😄

**Comment by esabol** [comment]
This PR is failing to pass the CI. Any idea why?

I think this pragma only works on gcc >= 8, but surely ubuntu-latest has gcc >= 8?

I think we'll need `#if __GNUC__ >= 8` put around these pragmas to continue to support older gccs. I really should get PR #334 in a state where it can be merged in order to test this on the full range of supported versions of gcc.

**Comment by SpamapS** [comment]
Actually I started this on 20.04 which is why it was called gcc8 but then I upgraded to 22.04 so now it's actually gcc11 for that specific one. ubuntu "latest" I think is still 20.04 and so gcc8 which is why the one failed.

**Comment by esabol** [comment]
> Actually I started this on 20.04 which is why it was called gcc8 but then I upgraded to 22.04 so now it's actually gcc11 for that specific one. ubuntu "latest" I think is still 20.04 and so gcc8 which is why the one failed.

Ah, OK. That makes sense!

File: libgearman/add.cc [hunk 2]
```diff
@@ -218,8 +218,12 @@ gearman_task_st *add_task(Client& client,
     {
       task->unique_length= GEARMAN_MAX_UNIQUE_SIZE -1; // Leave space for NULL byte
     }
-
+/* We check to make sure there's space for the NULL above so this
+   warning does not apply */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
```
Comment (esabol) [inline libgearman/add.cc:224]: I still think you need `#if __GNUC__ > 8` before this line?

File: libgearman/add.cc [hunk 3]
```diff
@@ -420,8 +424,12 @@ gearman_task_st *add_reducer_task(Client* client,
...
```
Comment (esabol) [inline libgearman/add.cc:430]: And another `#if __GNUC__ > 8` before this line...

**Review by esabol — CHANGES_REQUESTED** [review]

**Comment by SpamapS** [comment]
Thanks Ed. Added those guards. I agree it would be good to have GH actions for all the gcc's we support. This whole warnings as errors business is really tedious to support across so many distros. Might be good to think about dropping some old ones.

**Review by esabol — APPROVED** [review]

**Comment by esabol** [comment]
> Might be good to think about dropping some old ones.

Please don't say that after I spent so many, many hours getting the CI workflow to build with gcc 4.8 and 4.9. 😆

This is looking good. Two options on how to proceed:

(1) I remove the gcc-11 build from PR #334 and we merge that. Then you rebase PR #350 on the new master and add the gcc-11 build to the CI workflow in this PR.

(2) You merge PR #350 and then I rebase PR #334 on top of it. Less work, but it assumes PR #350 won't break any other builds.

Which option do you prefer?

**Comment by esabol** [comment]
Oh, it seems I'm too slow to comment. Never mind. Guess we are going with option 2. :smile:

**Comment by SpamapS** [comment]
I'm moving kind of fast as I'd like to push a release out today if
everything looks good. Using the new dockerfiles to test locally. :)

**Pull request merged**